### ISSUE TITLE
CID 73116: Incorrect deallocator used (CWE-762)

### ISSUE THREAD

**Issue by mmaraya** [mmaraya/netkit-ftp#32]
This new defect was introduced in https://github.com/mmaraya/netkit-ftp/commit/52c0978dcc1f18029907eb1f339338e437e039d8 when attempting to close #17.

[ftp/ftp.c#L513](https://github.com/mmaraya/netkit-ftp/blob/52c0978dcc1f18029907eb1f339338e437e039d8/ftp/ftp.c#L513)

```
alloc: Allocation of memory which must be freed using pclose.
assign: Assigning: fin = popen(local + 1, "r").
```

[ftp/ftp.c#L547](https://github.com/mmaraya/netkit-ftp/blob/52c0978dcc1f18029907eb1f339338e437e039d8/ftp/ftp.c#L547)

```
free: Calling fclose frees fin using fclose but it should have been freed using pclose.
```

Depending on the difference between the correct and incorrect allocator, there may be a resource leak or memory corruption. In sendrequest: A resource is freed using the wrong deallocator ([CWE-762](https://cwe.mitre.org/data/definitions/762.html))

**Comment by mmaraya** [comment]
Nope, still not fixed.

**Comment by mmaraya** [comment]
Fixed, but Coverity still reports #17 as open.
